instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/CopilotIntegrationTest.kt

```diff
@@ -0,0 +1,148 @@
+import com.mapbox.navigation.testing.ui.BaseTest
+import com.mapbox.navigation.testing.ui.utils.getMapboxAccessTokenFromResources
+import com.mapbox.navigation.testing.ui.utils.runOnMainSync
+import com.mapbox.navigation.utils.internal.logE
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.first
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class CopilotIntegrationTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.java) {
+
+    private lateinit var mapboxNavigation: MapboxNavigation
+    private lateinit var mockRoute: MockRoute
+
+    @get:Rule
+    val mapboxNavigationRule = MapboxNavigationRule()
+    @get:Rule
+    val mapboxHistoryTestRule = MapboxHistoryTestRule()
+
+    override fun setupMockLocation(): Location {
+        mockRoute = RoutesProvider.dc_very_short(activity)
+        return mockLocationUpdatesRule.generateLocationUpdate {
+            latitude = mockRoute.routeWaypoints.first().latitude()
+            longitude = mockRoute.routeWaypoints.first().longitude()
+        }
+    }
+
+    @Before
+    fun setUp() {
+        runOnMainSync {
+            mapboxNavigation = MapboxNavigationProvider.create(
+                NavigationOptions.Builder(activity)
+                    .accessToken(getMapboxAccessTokenFromResources(activity))
+                    .build()
+            )
+            mapboxHistoryTestRule.historyRecorder = mapboxNavigation.historyRecorder
+        }
+    }
+
+    @Test
+    fun copilotObserverReceivesSameIdsAsDeveloperMetadata() = sdkTest {
+        val collectedDeveloperMetadatas = mutableListOf<DeveloperMetadata>()
+        val developerMetadataObserver = DeveloperMetadataObserver {
+            Log.i("[DeveloperMetadataObserver]", "collected metadata: $it")
+            collectedDeveloperMetadatas.add(it)
+        }
+        val startedSessionObserverIds = mutableListOf<String>()
+        val historyRecordingStateObserver = object : HistoryRecordingStateChangeObserver {
+            override fun onShouldStartRecording(state: NavigationSessionState) {
+                Log.i("[HistoryRecordingStateChangeObserver]", "start recording for $state")
+                startedSessionObserverIds.add(state.sessionId)
+            }
+            override fun onShouldStopRecording(state: NavigationSessionState) {
+            }
+
+            override fun onShouldCancelRecording(state: NavigationSessionState) {
+            }
+        }
+        mapboxNavigation.registerDeveloperMetadataObserver(developerMetadataObserver)
+        mockWebServerRule.requestHandlers.addAll(mockRoute.mockRequestHandlers)
+        mapboxNavigation.historyRecorder.startRecording()
+        mapboxNavigation.registerHistoryRecordingStateChangeObserver(historyRecordingStateObserver)
+
+        // Current: Idle
+        // Free Drive
```

Comment:
Wondering if we can move these comments into `private` `fun`ctions so the overall code is easier to read / understand and the comments not necessary anymore.

> The best comment is a good name for a method or class.